controllers/btpoperator_controller.go

```diff
@@ -1056,14 +1092,32 @@ func (r *BtpOperatorReconciler) watchSecretPredicates() predicate.Funcs {
 			if !ok {
 				return false
 			}
-			return secret.Name == SecretName && secret.Namespace == ChartNamespace
+			if secret.Name == SecretName && secret.Namespace == ChartNamespace {
```

Comment:
How about:

1. extracting this to separate function to avoid code duplication (see line 1111) and
2. using just logical expression:
``` return secret.Namespace == ChartNamespace && (secret.Name == SecretName || secretName == CaSecret || secret.Name == WebhookSecret) ```